source/FreeRTOS_TCP_IP.c

```diff
@@ -440,34 +467,55 @@
         if( ( eTCPState == eCLOSED ) ||
             ( eTCPState == eCLOSE_WAIT ) )
         {
+            BaseType_t xMustClear = pdFALSE;
+            BaseType_t xHasCleared = pdFALSE;
+
+            if( ( xParent == pxSocket ) && ( pxSocket->u.xTCP.pxPeerSocket != NULL ) )
+            {
+                xParent = pxSocket->u.xTCP.pxPeerSocket;
+            }
+
+            if( xParent != NULL )
```

Comment:
This NULL check seems unnecessary here, as there is an assertion for that [here](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1148/files#diff-cf3234a936ea0fcb8f1555c9bc8a8c8862edaa92db41a1ab43a244833c67b300L328).